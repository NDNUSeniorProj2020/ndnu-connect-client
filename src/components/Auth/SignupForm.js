import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Form, Input, Col, Row, Button, message } from 'antd';
import 'antd/dist/antd.css';

import validatePhoneNumber from '../../assets/js/validatePhoneNubmer';

const SignupForm = Form.create()(
  class extends Component {
    constructor(props){
      super(props);
      this.state = {
      confirmDirty: false,
      }
    }

    handleSubmit = e => {
      e.preventDefault();
      this.props.form.validateFieldsAndScroll((err, values) => {
        if (!err) {
          if (validatePhoneNumber(values.phone_number)) {
            const user = { user: { ...values } };
            this.props.handleSignup(user);
          } else {
            message.error('Enter a phone number in the format of (XXX) XXX-XXXX, (XXX)XXX-XXXX, or XXX-XXX-XXXX.');
          }
        }
      });
    };

    handleConfirmBlur = e => {
      const { value } = e.target;
      this.setState({ confirmDirty: this.state.confirmDirty || !!value });
    };

    compareToFirstPassword = (rule, value, callback) => {
      const { form } = this.props;
      if (value && value !== form.getFieldValue('password')) {
        callback('Two passwords that you enter is inconsistent!');
      } else {
        callback();
      }
    };

    validateToNextPassword = (rule, value, callback) => {
      const { form } = this.props;
      if (value && this.state.confirmDirty) {
        form.validateFields(['confirm'], { force: true });
      }
      callback();
    };

    render() {
      const { getFieldDecorator } = this.props.form;
      const tailFormItemLayout = {
        wrapperCol: {
          xs: {
            span: 24,
            offset: 0,
          },
          sm: {
            span: 16,
            offset: 8,
          },
        },
      };

      return (
        <div>
          <h1 style={{textAlign:"left"}}>Sign Up</h1>
          <Form style={{width:"500px"}} layout='vertical' onSubmit={this.handleSubmit}>
            <Row>
            <Col span={12} key={1} style={{paddingRight:"10px"}}>
            <Form.Item label="First Name">
              {getFieldDecorator('first_name', {
                rules: [
                  {
                    required: true,
                    message: 'Please input your first name!',
                  },
                ],
              })(<Input />)}
            </Form.Item>
            </Col>
            <Col span={12} key={2} style={{paddingRight:"10px"}}>
            <Form.Item label="Last Name">
              {getFieldDecorator('last_name', {
                rules: [
                  {
                    required: true,
                    message: 'Please input your last name!',
                  },
                ],
              })(<Input />)}
            </Form.Item>
            </Col>
            </Row>
            <Form.Item label="E-mail">
              {getFieldDecorator('email', {
                rules: [
                  {
                    type: 'email',
                    message: 'The input is not valid E-mail!',
                  },
                  {
                    required: true,
                    message: 'Please input your E-mail!',
                  },
                ],
              })(<Input />)}
            </Form.Item>
            <Form.Item label="Phone Number">
              {getFieldDecorator('phone_number', {
                rules: [
                  {
                    required: true,
                    message: 'Please enter a valid phone number!'
                  }
                ]
              })(<Input style={{ width: '100%' }} />)}
            </Form.Item>
            <Form.Item label="Password" hasFeedback>
              {getFieldDecorator('password', {
                rules: [
                  {
                    required: true,
                    message: 'Please input your password!',
                  },
                  {
                    validator: this.validateToNextPassword,
                  },
                ],
              })(<Input.Password />)}
            </Form.Item>
            <Form.Item label="Confirm Password" hasFeedback>
              {getFieldDecorator('confirm', {
                rules: [
                  {
                    required: true,
                    message: 'Please confirm your password!',
                  },
                  {
                    validator: this.compareToFirstPassword,
                  },
                ],
              })(<Input.Password onBlur={this.handleConfirmBlur} />)}
            </Form.Item>

            <Form.Item {...tailFormItemLayout}>
              <Button type="primary" htmlType="submit">
                Sign Up
              </Button>
            </Form.Item>
          </Form>
        </div>
      );
    }
  }
);

SignupForm.propTypes = { handleSignup: PropTypes.func };
SignupForm.defaultProps = { handleSignup: f => f };

export default SignupForm;
