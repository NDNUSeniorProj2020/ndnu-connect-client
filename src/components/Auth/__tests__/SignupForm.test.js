import React from 'react';
import renderer from 'react-test-renderer';

import SignupForm from '../SignupForm';

describe('Testing LoginForm component', () => {
	describe('Snapshot tests', () => {
		it('should render without crashing', () => {
			const tree = renderer.create(<SignupForm />);
			expect(tree).toMatchSnapshot();
		});
	});
});