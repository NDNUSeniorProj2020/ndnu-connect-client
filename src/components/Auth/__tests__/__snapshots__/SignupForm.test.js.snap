// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for SignupForm snapshot tests should render without crashing 1`] = `
<div>
  <h1
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    Sign Up
  </h1>
  <form
    className="ant-form ant-form-vertical"
    onSubmit={[Function]}
    style={
      Object {
        "width": "500px",
      }
    }
  >
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-col ant-col-12"
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className="ant-form-item-required"
              htmlFor="first_name"
              onClick={[Function]}
              title="First Name"
            >
              First Name
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control-wrapper"
            style={Object {}}
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  data-__field={
                    Object {
                      "name": "first_name",
                    }
                  }
                  data-__meta={
                    Object {
                      "name": "first_name",
                      "originalProps": Object {
                        "type": "text",
                      },
                      "ref": null,
                      "rules": Array [
                        Object {
                          "message": "Please input your first name!",
                          "required": true,
                        },
                      ],
                      "trigger": "onChange",
                      "validate": Array [
                        Object {
                          "rules": Array [
                            Object {
                              "message": "Please input your first name!",
                              "required": true,
                            },
                          ],
                          "trigger": Array [
                            "onChange",
                          ],
                        },
                      ],
                      "valuePropName": "value",
                    }
                  }
                  id="first_name"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-col ant-col-12"
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className="ant-form-item-required"
              htmlFor="last_name"
              onClick={[Function]}
              title="Last Name"
            >
              Last Name
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control-wrapper"
            style={Object {}}
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  data-__field={
                    Object {
                      "name": "last_name",
                    }
                  }
                  data-__meta={
                    Object {
                      "name": "last_name",
                      "originalProps": Object {
                        "type": "text",
                      },
                      "ref": null,
                      "rules": Array [
                        Object {
                          "message": "Please input your last name!",
                          "required": true,
                        },
                      ],
                      "trigger": "onChange",
                      "validate": Array [
                        Object {
                          "rules": Array [
                            Object {
                              "message": "Please input your last name!",
                              "required": true,
                            },
                          ],
                          "trigger": Array [
                            "onChange",
                          ],
                        },
                      ],
                      "valuePropName": "value",
                    }
                  }
                  id="last_name"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="email"
          onClick={[Function]}
          title="E-mail"
        >
          E-mail
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control-wrapper"
        style={Object {}}
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <input
              className="ant-input"
              data-__field={
                Object {
                  "name": "email",
                }
              }
              data-__meta={
                Object {
                  "name": "email",
                  "originalProps": Object {
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "The input is not valid E-mail!",
                      "type": "email",
                    },
                    Object {
                      "message": "Please input your E-mail!",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "The input is not valid E-mail!",
                          "type": "email",
                        },
                        Object {
                          "message": "Please input your E-mail!",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              id="email"
              onChange={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="phone_number"
          onClick={[Function]}
          title="Phone Number"
        >
          Phone Number
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control-wrapper"
        style={Object {}}
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <input
              className="ant-input"
              data-__field={
                Object {
                  "name": "phone_number",
                }
              }
              data-__meta={
                Object {
                  "name": "phone_number",
                  "originalProps": Object {
                    "style": Object {
                      "width": "100%",
                    },
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please enter a valid phone number!",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please enter a valid phone number!",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              id="phone_number"
              onChange={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="password"
          onClick={[Function]}
          title="Password"
        >
          Password
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control-wrapper"
        style={Object {}}
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-password ant-input-affix-wrapper"
            >
              <input
                action="click"
                className="ant-input"
                data-__field={
                  Object {
                    "name": "password",
                  }
                }
                data-__meta={
                  Object {
                    "name": "password",
                    "originalProps": Object {
                      "action": "click",
                      "inputPrefixCls": "ant-input",
                      "prefixCls": "ant-input-password",
                      "visibilityToggle": true,
                    },
                    "ref": null,
                    "rules": Array [
                      Object {
                        "message": "Please input your password!",
                        "required": true,
                      },
                      Object {
                        "validator": [Function],
                      },
                    ],
                    "trigger": "onChange",
                    "validate": Array [
                      Object {
                        "rules": Array [
                          Object {
                            "message": "Please input your password!",
                            "required": true,
                          },
                          Object {
                            "validator": [Function],
                          },
                        ],
                        "trigger": Array [
                          "onChange",
                        ],
                      },
                    ],
                    "valuePropName": "value",
                  }
                }
                id="password"
                onChange={[Function]}
                onKeyDown={[Function]}
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <i
                  aria-label="icon: eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                    />
                  </svg>
                </i>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="confirm"
          onClick={[Function]}
          title="Confirm Password"
        >
          Confirm Password
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control-wrapper"
        style={Object {}}
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-password ant-input-affix-wrapper"
            >
              <input
                action="click"
                className="ant-input"
                data-__field={
                  Object {
                    "name": "confirm",
                  }
                }
                data-__meta={
                  Object {
                    "name": "confirm",
                    "originalProps": Object {
                      "action": "click",
                      "inputPrefixCls": "ant-input",
                      "onBlur": [Function],
                      "prefixCls": "ant-input-password",
                      "visibilityToggle": true,
                    },
                    "ref": null,
                    "rules": Array [
                      Object {
                        "message": "Please confirm your password!",
                        "required": true,
                      },
                      Object {
                        "validator": [Function],
                      },
                    ],
                    "trigger": "onChange",
                    "validate": Array [
                      Object {
                        "rules": Array [
                          Object {
                            "message": "Please confirm your password!",
                            "required": true,
                          },
                          Object {
                            "validator": [Function],
                          },
                        ],
                        "trigger": Array [
                          "onChange",
                        ],
                      },
                    ],
                    "valuePropName": "value",
                  }
                }
                id="confirm"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <i
                  aria-label="icon: eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                    />
                  </svg>
                </i>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Sign Up
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
